flink/v1.12/flink/src/main/java/com/netease/arctic/flink/write/ArcticFileWriter.java

```diff
@@ -198,7 +198,17 @@ public void dispose() throws Exception {
   }
 
   private void emit(WriteResult writeResult) {
-    output.collect(new StreamRecord<>(writeResult));
+    if (!isWriteResultEmpty(writeResult)) {
+      // Only emit a non-empty WriteResult to committer operator, thus avoiding submitting too much empty snapshots.
```

Comment:
It's better to give a comment about the situation hoe the empty `WriteResult` produces.